pkg/controllers/disruption/emptynodeconsolidation.go

```diff
@@ -52,20 +52,29 @@ func (c *EmptyNodeConsolidation) ComputeCommand(ctx context.Context, disruptionB
 	}).Set(float64(len(candidates)))
 
 	empty := make([]*Candidate, 0, len(candidates))
+	constrainedByBudgets := false
 	for _, candidate := range candidates {
 		if len(candidate.pods) > 0 {
 			continue
 		}
+		if disruptionBudgetMapping[candidate.nodePool.Name] == 0 {
+			// set constrainedByBudgets to true if any node was a candidate but was constrained by a budget
+			constrainedByBudgets = true
```

Comment:
If we ever can't remove a candidate due to a budget constraint, we should fire an event?

Reply:
We did something similar for NodePool here https://github.com/kubernetes-sigs/karpenter/pull/960. I think we can similarly add this for a candidate in the future too. One thing to consider is let's say a budget allows 3 disruptions, but we have 5 candidates. Do we emit an event for all 5 of the candidates, or just a random 2? This is probably trivial to do for Expiration and Drift, and we can just probably emit, but for multiNodeConsolidation, where the correctness of the command depends on how it's batched, this may make less sense.